EventsExpress.Core/Services/TokenService.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IdentityModel.Tokens.Jwt;
+using System.Linq;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using EventsExpress.Core.DTOs;
+using EventsExpress.Core.Infrastructure;
+using EventsExpress.Core.IServices;
+using EventsExpress.Db.Entities;
+using Microsoft.Extensions.Configuration;
+using Microsoft.IdentityModel.Tokens;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using AutoMapper;
+
+namespace EventsExpress.Core.Services
+{
+    public class TokenService : ITokenService
+    {
+        private readonly IUserService _userService;
+        private readonly IJwtSigningEncodingKey _signingEncodingKey;
+        private readonly IConfiguration _configuration;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IMapper _mapper;
+
+
+        public TokenService(IConfiguration configuration, IJwtSigningEncodingKey jwtSigningEncodingKey, IUserService userService, IHttpContextAccessor httpContextAccessor, IMapper mapper)
+        {
+            _configuration = configuration;
+            _signingEncodingKey = jwtSigningEncodingKey;
+            _userService = userService;
+            _httpContextAccessor = httpContextAccessor;
+            _mapper = mapper;
+        }
+        public string GenerateAccessToken(UserDTO user)
+        {
+            var lifeTime = _configuration.GetValue<int>("JWTOptions:LifeTime");
+            var claims = new[]
+            {
+                new Claim(ClaimTypes.Email, user.Email),
+                new Claim(ClaimTypes.Role, user.Role.Name),
+                new Claim(ClaimTypes.Name, user.Id.ToString()),
+            };
+            var jwtToken = new JwtSecurityToken(
+                claims: claims,
+                expires: DateTime.Now.AddSeconds(lifeTime),
+                signingCredentials: new SigningCredentials(
+                        _signingEncodingKey.GetKey(),
+                        _signingEncodingKey.SigningAlgorithm)
+            );
+            return new JwtSecurityTokenHandler().WriteToken(jwtToken);
+        }
+        public RefreshToken GenerateRefreshToken(string ipAddress)
```

Comment:
1. You do not need to pass IP here - you can retrieve it by call this.IPAddress
2. It would be better to have smth like "public RefreshToken GenerateRefreshToken(UserDTO user)" as for JWT, so you do not need remember about save token logic outside TokenService

Reply:
1. did it